### PR TITLE
Fix anchors being disabled in obsolete Adaptive Cards

### LINKED ISSUE #3582: [Webchat modification required] : https://github.com/microsoft/BotFramework-WebChat/tree/master/samples/05.custom-components/l.disable-adaptive-cards
I was working on one repro and encountered that the : https://github.com/microsoft/BotFramework-WebChat/tree/master/samples/05.custom-components/l.disable-adaptive-cards has an extra line of code added on line 205 :               disabled={disabled}

## Version

> To determine what version of Web Chat you are running, open your browser's development tools, and paste the following line of code into the console.
>
> `[].map.call(document.head.querySelectorAll('meta[name^="botframework-"]'), function (meta) { return meta.outerHTML; }).join('\n')`
>
> If you are using Web Chat outside of a browser, please specify your hosting environment. For example, React Native on iOS, Cordova on Android, SharePoint, PowerApps, etc.

## Steps to reproduce

1. Go to https://github.com/microsoft/BotFramework-WebChat/blob/master/samples/05.custom-components/l.disable-adaptive-cards/index.html
1. try to run the sample
1. we will see error 
1. post removing code from line 205 : disabled={disabled}
It will work 

It do not work with the code where we have  disabled={disabled} added under the ReactWebChat

## Expected behavior

provide a response 

## Additional context

[Bug]

### LINKED ISSUE #3618: Adaptive Card links should not be disabled even if the card is disabled
## Version
4.11.0

## Describe the bug

Currently, published behavior is that Web Chat's anchors will not work on first click, but will redirect to the url on second click. The original intention was to disable all element clicks, not just the first one. This behavior needs to be fixed.

When Adaptive Cards are disabled after use (when they are no longer the most recent activity), buttons and inputs should be disabled, but anchors should not. 

Right now the AC renderer will disable the following elements: `<a>`, `<button>`, `<input />`, `<select>`, `<textarea>`. All elements except anchors are reasonable to disable, but anchors should still be accessible even if the rest of the card is not. Since all anchors in Adaptive Cards are treated as links, as opposed to buttons, it's superfluous to disable them.

This bug is two-fold:
1.  Disabling the anchor should be removed
1. The event listener undoing the action for other elements should listen on ALL clicks, not just the first one.

## Steps to reproduce
1. Show a card that contains an anchor
1. Send another activity
1. Click on anchor
1. First click does nothing
1. Second click navigates to url

`attachmentMiddleware`:
```
      var attachmentMiddleware = function () {
        return function (next) {
          return function (card) {
            var attachment = card.attachment;
 
            switch (attachment.contentType) {
              case 'application/vnd.microsoft.card.adaptive':
                return window.React.createElement(window.WebChat.Components.AdaptiveCardContent, {
                  actionPerformedClassName: 'card__action--performed',
                  content: attachment.content,
                  disabled: true
                  // disabled: !recentBotMessage
                });
 
              default:
                return next(card);
            }
          };
        };
      };
```

## Expected behavior
Anchors should not be disabled, and disabled elements's event listener with undo should happen on every subsequent click, not just on the first click.

1. Remove anchors from disabled elements list
1. Add `addEventListenerWithUndo` that listens to all clicks, not just the first (as opposed to `addEventListenerOnceWithUndo`

### Addtional context

Testing scenarios:
1. With AC disabled, make sure links is still working and opening new window
1. Add a new test
   1. Add a new attachment middleware that disable AC, but not hero card
   1. Put a document.onclick handler
   1. Type "card breakfast"
   1. With AC disabled, click on one of those disabled button
   1. Expect, the document.onclick handler is not fired
   1. Type "herocard qna"
   1. Click on one of the button on hero card
   1. Expect, the document.onclick handler is fired once

[Bug]

### PR DESCRIPTION
**Pull request by corinagum**
> Fixes #3618 
> Fixes #3582

## Description

Previously, Adaptive Cards that have been marked as obsolete in the DOM were disabling anchors, which is invalid behavior. This change makes it so that while other inputs are disabled, anchors remain enabled and can be clicked/navigated to when in an older message in the transcript.

This also fixes the Disable Adaptive Cards sample, which had an error in it.

-  [x] I have added tests and executed them locally
-  [x] I have updated `CHANGELOG.md`
-  ~[ ] I have updated documentation~

## Review Checklist

> This section is for contributors to review your work.

-  [ ] Accessibility reviewed (tab order, content readability, alt text, color contrast)
-  [ ] Browser and platform compatibilities reviewed
-  [ ] CSS styles reviewed (minimal rules, no `z-index`)
-  [ ] Documents reviewed (docs, samples, live demo)
-  [ ] Internationalization reviewed (strings, unit formatting)
-  [ ] `package.json` and `package-lock.json` reviewed
-  [ ] Security reviewed (no data URIs, check for nonce leak)
-  [ ] Tests reviewed (coverage, legitimacy)

### CODE REVIEW

File: packages/bundle/src/adaptiveCards/Attachment/AdaptiveCardRenderer.js [hunk 1]
```diff
@@ -84,19 +84,19 @@ function addEventListenerOnceWithUndo(element, name, handler) {
...
+
 function disableElementWithUndo(element) {
   const undoStack = [];
   const isActive = element === document.activeElement;
   const tag = element.nodeName.toLowerCase();
 
   /* eslint-disable-next-line default-case */
   switch (tag) {
-    // Should we not disable <a>? Will some of the <a> are styled as button?
-    case 'a':
```
Comment (amal-khalaf) [inline packages/bundle/src/adaptiveCards/Attachment/AdaptiveCardRenderer.js:95]: should we also  remove 'a' on line 134 ?

File: packages/bundle/src/adaptiveCards/Attachment/AdaptiveCardRenderer.js [hunk 2]
```diff
@@ -84,19 +84,19 @@ function addEventListenerOnceWithUndo(element, name, handler) {
...
 }
 
+function addEventListenerWithUndo(element, name, handler) {
+  element.addEventListener(name, handler);
+
+  return () => element.removeEventListener(name, handler);
+}
+
 function disableElementWithUndo(element) {
   const undoStack = [];
```
Comment (amal-khalaf) [inline packages/bundle/src/adaptiveCards/Attachment/AdaptiveCardRenderer.js:94]: just wondering why do we call it **undo**stack?

File: packages/playground/src/App.js [hunk 3]
```diff
@@ -57,6 +68,11 @@ function App() {
 
   const [messageActivityWordBreak, setMessageActivityWordBreak] = useState('break-word');
 
+  // 'middlewareDisableObsoleteAC': Middleware that renders inputs disabled on an Adaptive Card that is no longer the latest activity.
+  const [middlewareDisableObsoleteAC, setMiddlewareDisableObsoleteAC] = useState(
```
Comment (amal-khalaf) [inline packages/playground/src/App.js:72]: nit: rename  to DisableObsoleteAC

File: packages/playground/src/App.js [hunk 4]
```diff
@@ -185,14 +233,14 @@ function App() {
...
     (e, checked) => {
-      setSendTypingIndicator(!!checked);
+      setSendTypingIndicator(checked);
     },
     [setSendTypingIndicator]
   );
 
   const handleUIDisabledChange = useCallback(
     (e, checked) => {
-      setDisabledUI(!!checked);
```
Comment (amal-khalaf) [inline packages/playground/src/App.js:195]: why did we remove `!!`

File: packages/playground/src/App.js [hunk 5]
```diff
@@ -21,8 +21,19 @@ function App() {
   const REDUX_STORE_KEY = 'REDUX_STORE';
 
   // fluentui settings
-  const stackTokens = { childrenGap: 10 };
-
+  const stackTokens = { childrenGap: 0 };
+
+  // TODO: Add action buttons in tooltip later
+  const wordBreakTooltipProps = {
```
Comment (amal-khalaf) [inline packages/playground/src/App.js:27]: it is not clear what is this WordBreakToolTip do could we add  2 liners  what is this doing/testing?

**Review by amal-khalaf — CHANGES_REQUESTED** [review]
pending comments

File: packages/bundle/src/adaptiveCards/Attachment/AdaptiveCardRenderer.js (hunk 1 above)
Comment (corinagum) [reply packages/bundle/src/adaptiveCards/Attachment/AdaptiveCardRenderer.js:95]: yeah, thanks for the catch

File: packages/bundle/src/adaptiveCards/Attachment/AdaptiveCardRenderer.js (hunk 2 above)
Comment (corinagum) [reply packages/bundle/src/adaptiveCards/Attachment/AdaptiveCardRenderer.js:94]: not sure - would have to ask compulim

File: packages/playground/src/App.js (hunk 5 above)
Comment (corinagum) [reply packages/playground/src/App.js:27]: this is the renderer for that tooltip, since it needs jsx. Not sure what you mean by 2 liners?

**Review by amal-khalaf — APPROVED** [review]

File: packages/playground/src/App.js (hunk 4 above)
Comment (corinagum) [reply packages/playground/src/App.js:195]: I already have falsy checks in state or session storage, plus I think it was confusing me during testing. Do you prefer keeping it? I can revert and I'll test to verify behavior if yes.

**Pull request merged**